cmd/influx/write.go

```diff
@@ -74,41 +90,102 @@ func cmdWrite(f *globalFlags, opt genericCLIOpts) *cobra.Command {
 	return cmd
 }
 
+func (writeFlags *writeFlagsType) dump(args []string) {
+	if writeFlags.Debug {
+		writeFlags.args = args
+		log.Printf("WriteFlags%+v", *writeFlags)
+	}
+}
+
 // createLineReader uses writeFlags and cli arguments to create a reader that produces line protocol
-func (writeFlags *writeFlagsType) createLineReader(args []string) (r io.Reader, closer io.Closer, err error) {
-	if len(args) > 0 && args[0][0] == '@' {
+func (writeFlags *writeFlagsType) createLineReader(cmd *cobra.Command, args []string) (io.Reader, io.Closer, error) {
+	readers := make([]io.Reader, 0, 2*len(writeFlags.Headers)+2*len(writeFlags.Files)+1)
+	closers := make([]io.Closer, 0, len(writeFlags.Files))
+
+	if len(args) > 0 && len(args[0]) > 1 && args[0][0] == '@' {
 		// backward compatibility: @ in arg denotes a file
-		writeFlags.File = args[0][1:]
+		writeFlags.Files = append(writeFlags.Files, args[0][1:])
```

Comment:
I'm concerned the flags are being used as a bucket for a bunch of side effects here

Reply:
yes, for the sake of backward compatibility @file means a file ... somebody introduced this and I decided to keep this functionality because I don't know where and how it is currently used. When starting from scratch, I would not allow specifying data in an argument at all.

Reply:
I don't htink my eariler comment was clear. I'd rather see you maintain the files in a sep variable instead of creating a destructive call on the flags. Flags should only be used (imo), for the API contract between CLI and user. Makes things coupled in weird global ways atm.

Reply:
understood, repaired